docs(portal): update structure and copy for templating/files docs (#1568)
Kong/docs.konghq.com: app/enterprise/1.3-x/developer-portal/working-with-templates.md

  </body>
</html>
```
{% endraw %}

### collections
Collections are a powerful tool enabling you to render sets of content as a group.  Content rendered as a collection share a configurable route pattern, as well as a layout.  Collections are configured in you portals `portal.conf.yaml` file.

The example below shows all the necessary configuration/files needed to render a basic `blog` collection made up of individual `posts`.

#### portal.conf.yaml

{% raw %}
```
name: Kong Portal
theme:
  name: base
collections:
  posts:
    output: true
    route: /:stub/:collection/:name
    layout: post.html
```
{% endraw %}

> Above you can see we have declared a `collections` object which is made up of individual collection configurations.  In this example we are configuring a collection called `posts`.  The renderer will look for a root directory called `_posts` within the `content` folder for individual pages to render.  If we created another collection conf called `animals`, the renderer would look for a directory called `_animals` for content files to render.

Reviewer: ```suggestion
Above you can see we have declared a `collections` object which is made up of individual collection configurations.  In this example, we are configuring a collection called `posts`.  The renderer will look for a root directory called `_posts` within the `content` folder for individual pages to render.  If we created another collection conf called `animals`, the renderer would look for a directory called `_animals` for content files to render.
```

Each configuration item is made up of a few parts:
- ###### `output`
  - **required**: false
  - **type**: `boolean`

>   - **description**: This optional attribute detirmines whether the collections should render or not.  When set to `false` virtual routes for the collection will not be created.

Reviewer: ```suggestion
  - **description**: This optional attribute determines whether the collections should render or not.  When set to `false` virtual routes for the collection are not created.
```

- ###### `route`
  - **required**: true
  - **type**: `string`
  - **default**: `none`

>   - **description**: The `route` attribute is required and tells the renderer what pattern to generate collection routes from. A collection route should always include at least one valid dynamic namespace that will uniquely identify each collection member.

Reviewer: ```suggestion
  - **description**: The `route` attribute is required and tells the renderer what pattern to generate collection routes from. A collection route should always include at least one valid dynamic namespace that uniquely identifies each collection member.
```

    - Any namespace in the route declaration which begins with `:` is considered dynamic.
    - Only certain dynamic namespaces are recognized by kong as valid:
      - `:title`: Replaces namespac with a contents `title`, declared in headmatter.
      - `:name`: Replaces namespace with the filename of a piece of content.
      - `:collection`: Replaces namespace with name of current collection.
      - `:stub`: Replaces namespace with value of `headmatter.stub` in each contents headmatter.
- ###### `route`
    - **required**: true
      - **type**: `boolean`

>       - **description**: The `layout` attribute determines what html layout the collections will use to render.  The path root is assessed from within the current themes `layouts` directory.

Reviewer: ```suggestion
      - **description**: The `layout` attribute determines what HTML layout the collections will use to render.  The path root is assessed from within the current themes `layouts` directory.
```
